### PR TITLE
feat: add C implementation for `stats/base/dists/kumaraswamy/median`

### LINKED ISSUE #3701: [RFC]: Add C implementation for `@stdlib/stats/base/dists/kumaraswamy/median`
### Description

This RFC proposes adding a C implementation, including a Node.js native addon, C benchmarks, and C examples, for the [`@stdlib/stats/base/dists/kumaraswamy/median`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median) package.

When adding support, the following tasks should be completed:

1. A C source implementation should be added to a `src` folder within the package's directory.
2. A native add-on interface should be added in order to call the C implementation from JavaScript.
3. A JavaScript file should be added to the `lib` folder to allow benchmarking and unit testing the native implementation from JavaScript.
4. A new test file should be created which includes unit tests testing against expected behavior.
5. C benchmarks should be added which measure performance for the C source implementation.
6. A file containing C examples should be added to demonstrate example usage.
7. JavaScript benchmarks should be added which measure performance for the native add-on.
8. Build and configuration files should be added to allow compiling and running the C source implementation using project `make` commands (as documented below).
9. The README should be updated to include documentation for the added C API.

To provide a concrete example of what a PR adding the desired method should contain, see https://github.com/stdlib-js/stdlib/pull/3354, which is a PR adding a C implementation for the CDF of an arcsine distribution. This should provide an idea of what is expected.

#### Prerequisites

1. Review JavaScript implementation in `lib/main.js` and its required modules to identify all `stdlib` functions used.
2. Verify C implementations exist for each required function.

### Related Issues

#3355

### Questions

No.

### Other

Once the implementation is ready including C examples and benchmarks, we should be able to run the following `make` commands.

#### Build native add-on

```bash
NODE_ADDONS_PATTERN="@stdlib/stats/base/dists/kumaraswamy/median" make install-node-addons
```

#### Run C examples

```bash
make examples-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/examples/c/example.c"
```

#### Run C benchmarks

```bash
make benchmark-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/benchmark/c/benchmark.c"
```

#### Run JavaScript benchmarks

```bash
make benchmark-javascript-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/benchmark/benchmark.native.js"
```

#### Run all tests

```bash
make TESTS_FILTER=".*/@stdlib/stats/base/dists/kumaraswamy/median/.*" test
```

### Notes

- If you are interested in contributing a PR which addresses this RFC and are a **first-time contributor** or not yet familiar with our project conventions, please minimize the submission of LLM-generated code. Please consult our [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md) and the associated [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/development.md). Failure to respect project conventions will result in your PR being rejected without review. Thank you for understanding!

### Checklist

- [x] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.
- [x] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by DeshDeepakKant**
Resolves #3701 

## Description

> What is the purpose of this pull request?

This pull request:

-  adds C implementation for stats/base/dists/kumaraswamy/median along with relevant tests, docs, examples and benchmarks.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3701 
## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)

* * *

@stdlib-js/reviewers

### CODE REVIEW

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/**
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median.h]: This file needs to be removed.

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/README.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 @license Apache-2.0
 
-Copyright (c) 2018 The Stdlib Authors.
+Copyright (c) 2014 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/README.md]:
```suggestion
Copyright (c) 2018 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/README.md [hunk 3]
```diff
@@ -143,6 +143,114 @@ for ( i = 0; i < 10; i++ ) {
...
+The function accepts the following arguments:
+
+-   **a**: `[in] double` shape parameter.
+-   **b**: `[in] double` shape parameter.
+
+
+```c
+double stdlib_base_dists_kumaraswamy_median( const double a, const double b );
+
+```
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/README.md]:
````suggestion

```c
double stdlib_base_dists_kumaraswamy_median( const double a, const double b );
```
````

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/README.md [hunk 4]
```diff
@@ -143,6 +143,114 @@ for ( i = 0; i < 10; i++ ) {
...
+```c
+#include "stdlib/stats/base/dists/kumaraswamy/median.h"
+#include "stdlib/constants/float64/eps.h"
+#include <stdlib.h>
+#include <stdio.h>
+
+double stdlib_base_dists_kumaraswamy_median( const double a, const double b ) {
+    return pow( 1.0 - pow( 0.5, 1.0 / b ), 1.0 / a );
+}
+
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/README.md]: We don't want to inline the implementation here.
```suggestion
```

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/src/main.c [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/stats/base/dists/kumaraswamy/median.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include <math.h>
+
+
+/**
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/src/main.c]:
```suggestion
#include <math.h>

/**
```

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/package.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@stdlib/stats/base/dists/kumaraswamy/median",
-  "version": "0.0.0",
-  "description": "Kumaraswamy's double bounded distribution median.",
+  "version": "0.0.1",
+  "description": "Kumaraswamy distribution median.",
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/package.json]: This file contains lots of unwanted changes that need to be reverted.

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/package.json [hunk 7]
```diff
@@ -13,16 +13,24 @@
       "url": "https://github.com/stdlib-js/stdlib/graphs/contributors"
     }
   ],
-  "main": "./lib",
+  "main": "./lib/main.js",
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/package.json:16]: This shouldn't be changed.

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/manifest.json [hunk 8]
```diff
@@ -0,0 +1,79 @@
+{
+    "options": {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/manifest.json]: Indentation should be two spaces here.

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/manifest.json [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+        "libraries": [],
+        "libpath": [],
+        "dependencies": [
+          "@stdlib/math/base/assert/is-nan",
+          "@stdlib/constants/float64/eps"
+        ]
+      }
+    ]
+  }
+
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/manifest.json]: Ensure you have EditorConfig setup. Should have a trailing newline here and elsewhere.

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/lib/native.js [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Returns the median of a Kumaraswamy distribution.
+*
+* @param {number} a - shape parameter
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/lib/native.js]:
```suggestion
*
* @private
* @param {number} a - shape parameter
```

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/benchmark/c/benchmark.c [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+	double elapsed;
+	double a[ 100 ];
+	double b[ 100 ];
+	double y;
+	double t;
+	int i;
+
+	for ( i = 0; i < 100; i++ ) {
+		// Generate random parameters for Kumaraswamy distribution
+		// Ensure parameters are positive and within reasonable ranges
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/benchmark/c/benchmark.c]:
```suggestion
```

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/test/test.native.js [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+	y = median( 0.0, NINF );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	y = median( PINF, NINF );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	t.end();
+});
+
+tape( 'the function returns the median of a Cauchy distribution', opts, function test( t ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/test/test.native.js]:
```suggestion
tape( 'the function returns the median of a Kumaraswamy distribution', opts, function test( t ) {
```

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/test/fixtures/julia/runner.jl [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+
+``` julia
+julia> p = rand( 1000 );
+julia> gen( p, \"data.json\" );
+```
+"""
+function gen( p, name )
+	z = Array{Float64}( undef, length(p) );
+	for i in eachindex(p)
+		z[ i ] = var( Geometric( p[ i ] ) );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/test/fixtures/julia/runner.jl]: This is not the correct distribution.

**Review by Planeshifter — CHANGES_REQUESTED** [review]
Thanks for opening a PR!

Unfortunately, it right now contains a large number of errors that will have to be addressed before it can move forward.

**Comment by DeshDeepakKant** [comment]
Thanks for the review,

This is my first time contributing to open source,
I will fix it asap.

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/README.md [hunk 14]
```diff
@@ -143,6 +143,109 @@ for ( i = 0; i < 10; i++ ) {
...
+The function accepts the following arguments:
+
+-   **a**: `[in] double` shape parameter.
+-   **b**: `[in] double` shape parameter.
+
+```c
+double stdlib_base_dists_kumaraswamy_median( const double a, const double b );
+
+```
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/README.md]:
````suggestion
```
````

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/README.md [hunk 15]
```diff
@@ -143,6 +143,109 @@ for ( i = 0; i < 10; i++ ) {
...
+    for ( i = 0; i < 25; i++ ) {
+        a = random_uniform( 0.1, 10.0 );
+        b = random_uniform( 0.1, 10.0 );
+        y = stdlib_base_dists_kumaraswamy_median( a, b );
+        printf( "a: %lf, b: %lf, Median(a, b): %lf\n", a, b, y );
+    }
+
+    return 0;
+}
+
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/README.md:236]:
```suggestion
}
```

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/README.md [hunk 16]
```diff
@@ -143,6 +143,109 @@ for ( i = 0; i < 10; i++ ) {
...
+```
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/README.md]:
````suggestion
}
```
````

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/examples/c/example.c [hunk 17]
```diff
@@ -0,0 +1,45 @@
...
+
+int main( void ) {
+	double a;
+	double b;
+	double y;
+	int i;
+
+	for ( i = 0; i < 25; i++ ) {
+		// Generate random parameters for Kumaraswamy distribution
+		// Ensure parameters are positive and within reasonable ranges
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/examples/c/example.c]:
```suggestion
```

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/examples/c/example.c [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+		a = random_uniform( 0.0, 10.0 ) + STDLIB_CONSTANT_FLOAT64_EPS;
+		b = random_uniform( 0.0, 10.0 ) + STDLIB_CONSTANT_FLOAT64_EPS;
+
+		y = stdlib_base_dists_kumaraswamy_median( a, b );
+		printf( "a: %lf, b: %lf, Median(a,b): %lf\n", a, b, y );
+	}
+
+	return 0;
+}
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/examples/c/example.c]:
```suggestion
}

```

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/src/main.c [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/stats/base/dists/kumaraswamy/median.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include <math.h>
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/src/main.c]: We want to use our stdlib `pow`:

```suggestion
#include "stdlib/stats/base/dists/kumaraswamy/median.h"
#include "stdlib/math/base/assert/is_nan.h"
#include "stdlib/math/base/special/pow.h"
```

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/src/main.c [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+*/
+double stdlib_base_dists_kumaraswamy_median( const double a, const double b ) {
+    if (
+        stdlib_base_is_nan( a ) ||
+        stdlib_base_is_nan( b ) ||
+        a <= 0.0 || b <= 0.0
+    ) {
+        return 0.0/0.0; // NaN
+    }
+    return 1.0 - pow( 0.5, 1.0 / b );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/src/main.c]:
```suggestion
    return 1.0 - stdlib_base_pow( 0.5, 1.0 / b );
```

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/manifest.json [hunk 21]
```diff
@@ -0,0 +1,79 @@
...
+      ],
+      "include": [
+        "./include"
+      ],
+      "libraries": [],
+      "libpath": [],
+      "dependencies": [
+        "@stdlib/math/base/napi/binary",
+        "@stdlib/math/base/assert/is-nan",
+        "@stdlib/constants/float64/eps"
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/manifest.json]:
```suggestion
        "@stdlib/math/base/assert/is-nan",
        "@stdlib/math/base/special/pow",
        "@stdlib/constants/float64/eps"
```

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/manifest.json [hunk 22]
```diff
@@ -0,0 +1,79 @@
...
+        "./src/main.c"
+      ],
+      "include": [
+        "./include"
+      ],
+      "libraries": [],
+      "libpath": [],
+      "dependencies": [
+        "@stdlib/math/base/assert/is-nan",
+        "@stdlib/constants/float64/eps"
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/manifest.json]:
```suggestion
        "@stdlib/math/base/assert/is-nan",
        "@stdlib/math/base/special/pow",
        "@stdlib/constants/float64/eps"
```

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/manifest.json [hunk 23]
```diff
@@ -0,0 +1,79 @@
...
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/manifest.json]:
```suggestion
        "@stdlib/math/base/assert/is-nan",
        "@stdlib/math/base/special/pow",
        "@stdlib/constants/float64/eps"
```

File: lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/test/fixtures/julia/runner.jl [hunk 24]
```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env julia
+#
+# @license Apache-2.0
+#
+# Copyright (c) 2018 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/kumaraswamy/median/test/fixtures/julia/runner.jl]:
```suggestion
# Copyright (c) 2024 The Stdlib Authors.
```

**Comment by Planeshifter** [comment]
@DeshDeepakKant Thanks for all the changes! Did another review and, after a few more edits, I believe your PR should be ready to land now. This is an exciting milestone, as it marks your first contribution to open-source development. Let's go ahead and merge your contribution. 🚀

**Review by Planeshifter — APPROVED** [review]

**Pull request merged**